scripts/speech_recognition/code_switching/code_switching_manifest_creation.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import os
+import random
+
+# Checks -
+# Please normalize the text for each language so that we have only the alphabets (no numbers, special characters, punctuation)
+# Please ensure that the audio_fielpaths are absolute locations
+
+
+parser = argparse.ArgumentParser(description='Create synthetic code-switching data manifest from monolingual data')
+
+parser.add_argument("--manifest_language1", default=None, type=str, help='Manifest file for language 1')
+parser.add_argument("--manifest_language2", default=None, type=str, help='Manifest file for language 2')
+parser.add_argument(
+    "--manifest_save_path", default=None, type=str, help='Path to save created CS indermediate manifest'
+)
+parser.add_argument("--id_language1", default=None, type=str, help='Identifier for language 1, eg: en, es, hi')
+parser.add_argument("--id_language2", default=None, type=str, help='Identifier for language 2, eg: en, es, hi')
+parser.add_argument("--max_sample_duration_sec", default=19, type=int, help='Maximum duration of sample (sec)')
+parser.add_argument("--min_sample_duration_sec", default=16, type=int, help='Minimum duration of sample (sec)')
+parser.add_argument("--dataset_size_required_hrs", default=10, type=int, help='Duration of dataset required (hrs)')
+
+args = parser.parse_args()
+
+
+def __read_manifest(manifest_path: str):
+    """
+    Args:
+        manifest_path: absolute path of the manifest file
+
+    Returns:
+        List with manifest entires as elements
+
+    """
+    data = []
+
+    for line in open(manifest_path, 'r'):
+        data.append(json.loads(line))
+
+    return data
+
+
+def __write_manifest(manifest_path: str, data: list):
+    """
+    Args:
+        manifest_path: absolute path for where to save the manifest file
+        data: list consisting of entries for the manifest
+
+    Returns:
+
+    """
+
+    with open(manifest_path, 'w') as outfile:
+        for elem in data:
+            s = json.dumps(elem)
+            outfile.write(s + '\n')
+
+
+def __create_cs_manifest(
```

Comment:
i am not sure we need __ in scripts such as these since they are stand alone..

Reply:
fixed